test/components/popup.spec.tsx

```diff
@@ -0,0 +1,269 @@
+import React = require('react');
+import ReactDOM = require('react-dom');
+import {selectDom} from 'test-drive';
+import { ClientRenderer, expect, waitFor } from 'test-drive-react';
+import {PopupDemo} from '../../demo/components/popup-demo';
+import {HorizontalPosition, Popup, VerticalPosition} from '../../src/components/';
+
+const popup = 'POPUP';
+const container = 'POPUP_DEMO_DIV';
+
+describe('<Popup />', function() {
+    const clientRenderer = new ClientRenderer();
+    const bodySelect = selectDom(document.body);
+    const testContainer = document.body.appendChild(document.createElement('div'));
+
+    afterEach(function() {clientRenderer.cleanup(); });
+    after(() => {document.body.removeChild(testContainer); });
+
+    describe('The popup user', function() {
+        it('clicks on the parent and the popup opens and closes after another click', async function() {
+            const {select, waitForDom} = clientRenderer.render(<PopupDemo />);
+
+            await waitForDom(() => {
+                expect(select(container)).to.be.present();
+                expect(select(container, popup)).to.be.absent();
+            });
+            (select(container) as HTMLDivElement).click();
```

Comment:
select is a generic function. it accepts the target's type (extends `Element`). Use it instead of type casting.